library/ceph_key.py

```diff
@@ -0,0 +1,422 @@
+    '''
+    cmd = cmd
+
+    for k, v in caps.iteritems():
+        if type == "ceph-authtool":
+            cmd.extend(["--cap"])
+
+        cmd.extend([k, v])
+
+    return cmd
+
+
+def generate_ceph_cmd(cluster, args, containerized):
+    '''
+    Generate 'ceph' command line to execute
+    '''
+
+    cmd = []
+
+    base_cmd = [
+        'ceph',
+        '--cluster',
+        cluster,
+        'auth',
+    ]
+
+    cmd.extend(base_cmd + args)
+
+    if containerized:
+        cmd = containerized.split() + cmd
+
+    return cmd
+
+
+def generate_ceph_authtool_cmd(cluster, name, secret, caps, containerized):
+    '''
+    Generate 'ceph-authtool' command line to execute
+    '''
+
+    file_destination = "/etc/ceph/" + cluster + "." + name + ".keyring"
+
+    cmd = [
+        'ceph-authtool',
+        '--create-keyring',
+        file_destination,
+        '--name',
+        name,
+        '--add-key',
+        secret,
+    ]
+
+    cmd = generate_caps(cmd, "ceph-authtool", caps)
+
+    if containerized:
+        cmd = containerized.split() + cmd
+
+    return cmd
+
+
+def create_key(module, result, cluster, name, secret, caps, containerized):
+    '''
+    Create a CephX key
+    '''
+
+    file_path = "/etc/ceph/" + cluster + "." + name + ".keyring"
+
+    args = [
+        "import",
+        "-i",
+        file_path,
+    ]
+    cmd_list = []
+
+    if secret is None:
+        secret = generate_secret()
+
+    cmd_list.append(generate_ceph_authtool_cmd(cluster, name, secret, caps, containerized))
+    cmd_list.append(generate_ceph_cmd(cluster, args, containerized))
+
+    return cmd_list
+
+
+def update_key(cluster, name, caps, containerized):
+    '''
+    Update a CephX key's capabilities
+    '''
+
+    cmd_list = []
+
+    args = [
+        "caps",
+        name,
+    ]
+
+    args = generate_caps(args, "ceph", caps)
+    cmd_list.append(generate_ceph_cmd(cluster, args, containerized))
+
+    return cmd_list
+
+
+def delete_key(cluster, name, containerized):
+    '''
+    Delete a CephX key
+    '''
+
+    cmd_list = []
+
+    args = [
+        "del",
+        name,
+    ]
+
+    cmd_list.append(generate_ceph_cmd(cluster, args, containerized))
+
+    return cmd_list
+
+
+def info_key(cluster, name, containerized):
+    '''
+    Get information about a CephX key
+    '''
+
+    cmd_list = []
+
+    args = [
+        "get",
+        name,
+        '-f',
+        'json',
+    ]
+
+    cmd_list.append(generate_ceph_cmd(cluster, args, containerized))
+
+    return cmd_list
+
+
+def list_keys(cluster, containerized):
+    '''
+    List all CephX keys
+    '''
+
+    cmd_list = []
+
+    args = [
+        "ls",
+        '-f',
+        'json',
+    ]
+
+    cmd_list.append(generate_ceph_cmd(cluster, args, containerized))
+
+    return cmd_list
+
+
+def exec_commands(module, cmd_list):
+    '''
+    Execute command(s)
+    '''
+
+    for cmd in cmd_list:
+        rc, out, err = module.run_command(cmd, encoding=None)
```

Comment:
No encoding seems a bit dangerous here, is there any reason in particular why `None` was better than say `utf-8`?

Reply:
Nop, I guess I blindly took this from an existing module.